git-cliff-core/src/repo.rs

```diff
@@ -976,4 +1065,137 @@ mod test {
             assert!(!retain, "exclude: **/*.txt");
         }
     }
+
+    /// Test the semantic version comparison function
+    #[test]
+    fn test_semantic_version_compare() {
+        use std::cmp::Ordering;
+
+        // Test basic semantic version comparison
+        assert_eq!(
+            semantic_version_compare("v0.9.0", "v0.10.0"),
+            Ordering::Less
+        );
+        assert_eq!(
+            semantic_version_compare("v0.10.0", "v0.9.0"),
+            Ordering::Greater
+        );
+        assert_eq!(
+            semantic_version_compare("v1.0.0", "v1.0.0"),
+            Ordering::Equal
+        );
+
+        // Test without 'v' prefix
+        assert_eq!(semantic_version_compare("0.9.0", "0.10.0"), Ordering::Less);
+        assert_eq!(semantic_version_compare("1.2.3", "1.2.4"), Ordering::Less);
+
+        // Test with pre-release versions
+        assert_eq!(
+            semantic_version_compare("v1.0.0-alpha.1", "v1.0.0-alpha.2"),
+            Ordering::Less
+        );
+
+        // Test pre-release vs release (lines 574-575)
+        assert_eq!(
+            semantic_version_compare("v1.0.0-alpha.1", "v1.0.0"),
+            Ordering::Less
+        );
+        assert_eq!(
+            semantic_version_compare("v1.0.0", "v1.0.0-beta.1"),
+            Ordering::Greater
+        );
+
+        // Test different version lengths
+        assert_eq!(semantic_version_compare("v1.0", "v1.0.0"), Ordering::Less);
+        assert_eq!(
+            semantic_version_compare("v1.0.0", "v1.0"),
+            Ordering::Greater
+        );
+
+        // Test the specific case from the GitHub issue
+        assert_eq!(
+            semantic_version_compare("v0.9.0", "v0.10.0"),
+            Ordering::Less
+        );
+        assert_eq!(
+            semantic_version_compare("v0.10.0", "v0.11.0"),
+            Ordering::Less
+        );
```

Comment:
It's not clear which GitHub issue we are referring to here, it should be linked or the comment should be made more clear